src/ray/gcs/gcs_server/gcs_resource_manager.cc

```diff
@@ -15,19 +15,20 @@
 #include "ray/gcs/gcs_server/gcs_resource_manager.h"
 
 #include "ray/common/ray_config.h"
+#include "ray/gcs/gcs_server/ray_syncer.h"
 #include "ray/stats/metric_defs.h"
 
 namespace ray {
 namespace gcs {
 
 GcsResourceManager::GcsResourceManager(
     instrumented_io_context &main_io_service, std::shared_ptr<GcsPublisher> gcs_publisher,
-    std::shared_ptr<gcs::GcsTableStorage> gcs_table_storage)
+    std::shared_ptr<gcs::GcsTableStorage> gcs_table_storage,
+    syncer::RaySyncer *ray_syncer)
```

Comment:
@scv119  this will be removed in 4th PR. I make it a raw pointer so that I don't need to update unit tests for this one.

The specific component of poller/broadcaster is tested, but not integrated with this manager right now.

I'll leave it as it is for now. Once we created the good form of RaySyncer [within 2weeks].

I'll add test cases for that. For now, it's ok I think since it's not making things worse and we have nightly and CI to cover the errors.